src/sls_core.js

```diff
@@ -320,6 +320,60 @@ function getFunctionByName(functionName) {
   return new Function(`return ${functionName};`)();
 }
 
+/**
+ * Checks if the given URL is a valid image URL.
+ *
+ * @param {string} url The URL to check.
+ * @return {boolean} Whether the URL is a valid image URL.
+ */
+function isValidImageUrl(url) {
+  return url.startsWith('http://') || url.startsWith('https://');
+}
+
+/**
+ * Returns a file, which is assumed to be an image file, for a criteria client
+ * screenshot which should be named after a criteria id. Any formats are
+ * considered for the query, but it is assumed that the file will be an image.
+ *
+ * This file is retrieved from a folder created programmatically which is
+ * assumed to exist.
+ *
+ * If no such file has been found, the function returns the url of the default
+ * image mockup, which behaves analogously to an image file.
+ *
+ * There are warnings sent out (currently as a toast on the spreadsheet) in this
+ * case, and also in case that multiple image files are found. When finding
+ * multiple images, the last one is selected.
```

Comment:
There's no loop in the code. Doesn't this return the first image found?

Reply:
Eh, absolutely right! The second, I guess. That if should be a while.

Reply:
Wait, no. The first. It never meant to have a loop. 
(Did this a really long time ago)